mlrun/datastore/base.py

```diff
@@ -419,14 +421,34 @@ def as_df(
         :param df_module: optional, py module used to create the DataFrame (e.g. pd, dd, cudf, ..)
         :param format:    file format, if not specified it will be deducted from the suffix
         """
-        return self._store.as_df(
+        if "time_column" in kwargs or "time_field" in kwargs:
+            kwargs["time_column"] = kwargs.get("time_column") or kwargs.pop(
+                "time_field", None
+            )
+        df = self._store.as_df(
             self._url,
             self._path,
             columns=columns,
             df_module=df_module,
             format=format,
             **kwargs,
         )
+        if self._url.endswith(".json") or format == "json":
+            df = filter_df_start_end_time(
+                df,
+                time_column=kwargs.get("time_field"),
+                start_time=kwargs.get("start_time"),
+                end_time=kwargs.get("end_time"),
+            )
+            df = select_columns_from_df(df, columns=columns)
+        elif self._url.endswith(".csv") or format == "csv":
+            df = filter_df_start_end_time(
+                df,
+                time_column=kwargs.get("time_column"),
+                start_time=kwargs.get("start_time"),
+                end_time=kwargs.get("end_time"),
+            )
+        return df
```

Comment:
We only `select_columns_from_df` for json files, not csv files? It seems like this might be a code duplication accident. If so, please fix and refactor to avoid the duplication.

Reply:
if it is a csv file the column select executed in `as_df` func